channeldb/channel_test.go

```diff
@@ -651,6 +651,82 @@ func TestChannelStateTransition(t *testing.T) {
 	}
 }
 
+func TestFetchOpenChannel(t *testing.T) {
+	t.Parallel()
+
+	cdb, cleanUp, err := makeTestDB()
+	if err != nil {
+		t.Fatalf("unable to make test database: %v", err)
+	}
+	defer cleanUp()
+
+	// Create test channel state.
+	state, err := createTestChannelState(cdb)
+	if err != nil {
+		t.Fatalf("unable to create channel state: %v", err)
+	}
+
+	state.LocalCommitment.Htlcs = []HTLC{
+		{
+			Signature:     testSig.Serialize(),
+			Incoming:      true,
+			Amt:           10,
+			RHash:         key,
+			RefundTimeout: 1,
+			OnionBlob:     []byte("onionblob"),
+		},
+	}
+	state.RemoteCommitment.Htlcs = []HTLC{
+		{
+			Signature:     testSig.Serialize(),
+			Incoming:      false,
+			Amt:           10,
+			RHash:         key,
+			RefundTimeout: 1,
+			OnionBlob:     []byte("onionblob"),
+		},
+	}
+	if err := state.FullSync(); err != nil {
+		t.Fatalf("unable to save and serialize channel state: %v", err)
+	}
+
+	pubKey := state.IdentityPub.SerializeCompressed()
+	openChannel, err := cdb.FetchOpenChannel(pubKey, testOutpoint)
```

Comment:
error should be handled here, and if non-nil should be `Fatalf` for the test

Reply:
oo fixed